### PR TITLE
fix variable redeclared in error

### PR DESCRIPTION
**Pull request by paul-szczepanek-arm**
Decleration is shadowing the variable. The intention is to reuse the variable from outside the loop.

### CODE REVIEW

File: source/main.cpp [hunk 1]
```diff
@@ -184,7 +184,7 @@ class SocketDemo {
         /* loop until there is nothing received or we've ran out of buffer space */
         nsapi_size_or_error_t result = remaining_bytes;
         while (result > 0 && remaining_bytes > 0) {
```
Comment (paul-szczepanek-arm) [inline source/main.cpp:186]: not sure why we even bother checking result, there's an if in the loop that checks for errors

Comment (ghost) [reply source/main.cpp:186]: I guess we want to terminate the loop on error.

Comment (paul-szczepanek-arm) [reply source/main.cpp:186]: There's an if with an early return for that. I guess the difference is if  we got a zero sized packet but that would still be wrong if remaining bytes were >0

**Review by ghost — APPROVED** [review]

**Comment by paul-szczepanek-arm** [comment]
I'll look into the while condition but I'm merging this as clearly the shadowing is not intentional.

**Pull request merged**